main.cpp

```diff
@@ -250,65 +153,42 @@ FloatType& FloatType::divide(float rhs)
     return *this;
 }
 
-FloatType& FloatType::add(const FloatType& ft)
-{
-    return add(*ft.value);
-}
-
-FloatType& FloatType::add(const DoubleType& dt)
-{
-    return add(*dt.value);
-}
-
-FloatType& FloatType::add(const IntType& it)
-{
-    return add(*it.value);
-}
-
-FloatType& FloatType::subtract(const FloatType& ft)
-{
-    return subtract(*ft.value);
-}
-
-FloatType& FloatType::subtract(const DoubleType& dt)
-{
-    return subtract(*dt.value);
-}
-
-FloatType& FloatType::subtract(const IntType& it)
-{
-    return subtract(*it.value);
-}
-
-FloatType& FloatType::multiply(const FloatType& ft)
-{
-    return multiply(*ft.value);
-}
-
-FloatType& FloatType::multiply(const DoubleType& dt)
-{
-    return multiply(*dt.value);
-}
-
-FloatType& FloatType::multiply(const IntType& it)
+struct IntType
 {
-    return multiply(*it.value);
-}
+    IntType(int i_) : value(new int (i_)) {}
+    operator int () { return *value; }
+    ~IntType()
+    {
+        delete value;
+    }
```

Comment:
A coding style recommendation:
- Keep your constructors and destructors together.
- Keep your member functions together.

Your conversion function is written in between your constructor and destructor